### PR TITLE
Reinstate Static ActivityExtensions Constructor

### LINKED ISSUE #38283: [BUG] No opentelemetry traces are exported
### Library name and version

Azure.Messaging.ServiceBus 7.16.1

### Describe the bug

Using the following configuration

```csharp
builder.Services.AddOpenTelemetry()
            .WithTracing(tracerProviderBuilder =>
            {
                tracerProviderBuilder
                    .AddSource("Azure.*")
                    .SetSampler(new AlwaysOnSampler())
                    .AddAzureMonitorTraceExporter()
                    .AddConsoleExporter();
            })
builder.Services.AddAzureClients(configure =>
        {
            configure.AddBlobServiceClient("");
            configure.AddServiceBusClient("");
        });
```

I cannot observe any traces from the ServiceBus client within both exporters. However, I am able to see traces from the blob client.

I have registered both an ActivityListener and a DiagnosticListener. For the ServiceBus, I receive notifications from the DiagnosticListener but not from the ActivityListener. In the case of the Blob client, I can see traces from both sources.

### Expected behavior

Traces/Span from ServiceBus are exported like with `AddApplicationInsightsTelemetry`.

### Actual behavior

No traces

### Reproduction Steps

Simple aspnet core application that send messages with the telemetry configuration above

### Environment

- TargetFramework: net6.0
- Platform: local linux env under wsl2

### PR DESCRIPTION
**Pull request by m-redding**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/38283

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by lmolkova — APPROVED** [review]
Thank you!

File: sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs [hunk 1]
```diff
@@ -1043,6 +1043,11 @@ public static void ResetFeatureSwitch()
     /// </summary>
     internal static class ActivityExtensions
     {
+        static ActivityExtensions()
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs:1046]: I think we also need to update the check in RequestActivityPolicy to respect the flag.

Comment (m-redding) [reply sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs:1046]: Yep just added that as well

File: sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs [hunk 2]
```diff
@@ -187,13 +187,13 @@ private static ValueTask ProcessNextAsync(HttpMessage message, ReadOnlyMemory<Ht
 #if NETCOREAPP2_1
             (s_diagnosticSource.IsEnabled() || ActivityExtensions.ActivitySourceHasListeners(s_activitySource));
 #else
-            (s_diagnosticSource.IsEnabled() || s_activitySource.HasListeners());
+            (s_diagnosticSource.IsEnabled() || (s_activitySource.HasListeners() && ActivityExtensions.SupportsActivitySource));
```
Comment (JoshLove-msft) [inline sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs:190]: Can ShouldCreateActivity be simplified to use IsActivitySourceEnabled?

**Review by jsquire — APPROVED** [review]

File: sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs (hunk 2 above)
Comment (m-redding) [reply sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs]: updated

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**